src/get-or-create-render-context.js

```diff
@@ -7,53 +7,53 @@
 const vm = require("vm");
 // TODO(csilvers): try to get rid of the dependency on jsdom
 const jsdom = require("jsdom");
-const raf = require("raf");
 const cache = require("./cache.js");
 const profile = require("./profile.js");
 
-const runInContext = function(context, fn) {
-    return vm.runInContext("(" + fn.toString() + ")()", context);
-};
+class CustomResourceLoader extends jsdom.ResourceLoader {
+    fetch(url, options) {
+        // eslint-disable-next-line no-console
+        console.log("FETCHING: " + url);
+        return super.fetch(url, options);
+    }
+}
 
-const createRenderContext = function(jsPackages, pathToClientEntryPoint) {
-    const sandbox = {};
+const runInContext = function(jsdomContext, fnOrText) {
+    const script = typeof fnOrText === "function"
+        ? "(" + fnOrText.toString() + ")()"
+        : fnOrText;
+
+    if (typeof script !== "string") {
+        throw new Error("Must be a function or text");
+    }
+    return jsdomContext.runVMScript(new vm.Script(script));
+};
```

Comment:
Because jsdom now controls the vm context, we needed to change how we run code in context. I made the function take a function or text so we could use it for all script running and make our code a little easier to grok since they'll all call the same thing.